### PR TITLE
feat: Add restart support to the SSE client.

### PR DESCRIPTION
**Pull request by kinyoklion**
This adds restart support to the SSE client. We can then use this to restart the client on bad data and still include backoff/jitter.

### CODE REVIEW

File: apps/sse_contract_test_service/bin/sse_contract_test_service.dart [hunk 1]
```diff
@@ -7,14 +7,15 @@ import 'service_api.openapi.dart';
 import 'dart:io';
 
 class TestApiImpl extends TestApi {
-  static const clientUrlPrefix = "/client/";
+  static const _clientUrlPrefix = "/client/";
```
Comment (kinyoklion) [inline apps/sse_contract_test_service/bin/sse_contract_test_service.dart:10]: Changed these all to private variables. Unrelated to changes.

File: packages/event_source_client/lib/src/sse_client_html.dart [hunk 2]
```diff
@@ -49,6 +49,11 @@ class HtmlSseClient implements SSEClient {
 
   @override
   Future close() => _messageEventsController.close();
+
+  @override
+  void restart() {
+    // TODO: implement restart
```
Comment (kinyoklion) [inline packages/event_source_client/lib/src/sse_client_html.dart:55]: Not implementing this right now. We need to implement jitter/backoff for the web first.

**Review by louis-launchdarkly — APPROVED** [review]

**Review by tanderson-ld — APPROVED** [review]
